docs/contributor/arch/001-otel-collector-metric-based-pipeline-status.md

```diff
@@ -0,0 +1,102 @@
+Additionally, Prometheus automatically handles adjustments for breaks in monotonicity, such as counter resets resulting from target restarts. Manual scraping lacks this crucial feature.
+
+#### Prometheus as a Sidecar
+
+Now that we understand the necessity of using Prometheus for even basic metric analysis, let's explore how we can deploy it.
+One deployment approach involves deploying Prometheus as a sidecar alongside the operator.
+This method remains lightweight, as it doesn't necessitate any changes in the operator's code and can be easily achieved only using kustomize.
+However, this static setup does come with its drawbacks:
+
+* It relies on a static configuration with no room for parameterization, limiting flexibility.
+* Fine-tuning Prometheus dynamically based on the number of pipelines or other factors is not feasible.
+* Sidecar is deployed even when there are no pipelines, which may lead to unnecessary resource utilization.
+
+#### Standalone Prometheus Deployment
+
+An alternative approach would involve embedding Prometheus within the code of the operator, similar to how we deploy Fluent Bit or Otel Collector.
+This approach grants you complete control and maximum flexibility in managing Prometheus.
+It's equally important to communicate to the end user that this internal Prometheus is intended exclusively for internal purposes and should not be utilized for monitoring user workloads.
+Clear communication regarding its internal nature is crucial to avoid any misuses or misunderstandings.
+
+### Integrating Prometheus Querying into Operator Reconciliation Loop
+
+#### Direct Prometheus Queries:
+* The operator manages a set of predefined PromQL queries and communicates with Prometheus using [expression queries API](https://prometheus.io/docs/prometheus/latest/querying/api/#expression-queries) to collect data points.
+  Subsequently, it utilizes this data to evaluate the status of Pipeline CRs. 
+* However, it's crucial to note that if the querying process takes an extended duration, it may impact the reconciliation of TracePipeline and MetricPipeline.
+![Prometheus Integration using Direct Queries](../assets/prom-integration-direct-queries-flow.svg "Prometheus Integration using Direct Queries")
+
+#### Prometheus Queries with Additional Controller:
+* Similar to [Direct Prometheus Queries](#direct-prometheus-queries), but it addresses the potential challenge of long-running queries, which could hinder reconciliation introducing a separate PipelineMonitoring CR with its own reconciliation loop.
+  This separation effectively isolates the query execution and result storage from the TracePipeline and MetricPipeline controllers.
+  It's worth noting that while this approach enhances efficiency, it can introduce increased complexity to the overall setup.
+![Prometheus Integration with Additional Controller](../assets/prom-integration-extra-ctrl-flow.svg "Prometheus Integration with Additional Controller")
+
+#### Prometheus Alerts:
```

Comment:
As discussed: [recording rules](https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/) might be a more flexible alternative here.